modules/server/src/network/resolvers/connections/index.ts

```diff
@@ -0,0 +1,70 @@
+import { NetworkAggregationConfig, NetworkAggregationConfigInput } from '@/network/types';
+import { ObjectValues } from '@/utils/types';
+import axios from 'axios';
+import urljoin from 'url-join';
+
+/**
+ * Connection status types
+ */
+const CONNECTION_STATUS = {
+	OK: 'OK',
+	ERROR: 'ERROR',
+} as const;
+
+type ConnectionStatus = ObjectValues<typeof CONNECTION_STATUS>;
+
+/**
+ * Check the status of remote connections
+ *
+ * @param url - Remote connection url
+ * @returns A connection status
+ */
+const checkRemoteConnectionStatus = async (url: string): Promise<ConnectionStatus> => {
+	/**
+	 * recommended way to health check gql server is to run the `__typename` query that every server has
+	 * very small query with no additional params to, so using GET is not a concern for the GQL server
+	 * adds recommended pre flight header to make sure Apollo doesn't block request by CSRF protection
+	 */
+	const healthCheckQuery = '?query=%7B__typename%7D';
+	const fullUrl = urljoin(url, healthCheckQuery);
+
+	try {
+		const pong = await axios.get(fullUrl, {
+			headers: { 'apollo-require-preflight': 'true', 'Content-Type': 'application/json' },
+		});
+
+		// only need to check the server returns successfully (just a health check)
+		if (pong.data.data?.__typename) {
+			return CONNECTION_STATUS.OK;
+		} else {
+			throw Error('no data object returned from GQL server __typname query');
+		}
+	} catch (error) {
+		console.error(error);
+		return CONNECTION_STATUS.ERROR;
+	}
+};
+
+type RemoteConnectionData = NetworkAggregationConfigInput & {
+	availableAggregations: string[];
+	status: ConnectionStatus;
+};
+export const createRemoteConnectionResolvers = async (
+	networkConfigs: NetworkAggregationConfig[],
+): Promise<RemoteConnectionData[]> => {
```

Comment:
Is this actually returning a resolver (function that graphql server uses to collect requested data)? Its actually just returning the `RemoteConnectionData`. It might be named wrong or returning the wrong type.

Reply:
Because this returns a static object, it is only resolving the `status` value once, when `createRemoteConnectionResolvers` is first called. This means the `status` field is not providing any real value... we already know the remote arranger node was available at startup time because we include the node in our config. If we're going to report `status` in our node type then we should be pinging it at the time the gql request is made.